crates/core/src/db/datastore/locking_tx_datastore/committed_state.rs

```diff
@@ -233,20 +233,30 @@ impl CommittedState {
             with_label_values(ST_SEQUENCES_ID, ST_SEQUENCES_NAME).inc();
         }
 
+        self.reset_system_table_schemas(database_address)?;
+
+        Ok(())
+    }
+
+    /// Compute the system table schemas from the system tables,
+    /// and store those schemas in the in-memory [`Table`] structures.
+    ///
+    /// Necessary during bootstrap because system tables include autoinc IDs
+    /// for objects like indexes and constraints
+    /// which are computed at insert-time,
+    /// and therefore not included in the hardcoded schemas.
+    pub(super) fn reset_system_table_schemas(&mut self, database_address: Address) -> Result<()> {
         // Re-read the schema with the correct ids...
         let ctx = ExecutionContext::internal(database_address);
-        for table_id in ref_schemas.map(|s| s.table_id) {
-            let schema = self.schema_for_table_raw(&ctx, table_id)?;
-            self.tables
-                .get_mut(&table_id)
-                .unwrap()
-                .with_mut_schema(|ts| *ts = schema);
+        for schema in system_tables() {
+            self.tables.get_mut(&schema.table_id).unwrap().schema =
+                Arc::new(self.schema_for_table_raw(&ctx, schema.table_id)?);
```

Comment:
This is the way it was written in `bootstrap_system_tables`; I copied the body of this function from there.

Reply:
Unrelatedly, I don't really understand what this is doing or what the point of it is really. I don't recall in what ways the system tables change from original hardcoded schemas on startup. So I don't really understand why this needs to change

Reply:
There is no reason not to use `.with_mut_schema` as far as I am aware, but I did not go that route because I did not change this code.

What's happening is that the hardcoded schemas have 0 for all the autoinc IDs of columns and indices and constrants, AFAICT because we don't want to manually assign and maintain IDs for all the objects in the metaschemas. So when you initially install the `struct Schema` in the `struct Table` from the hardcoded stuff in `system_tables.rs`, you have zeros in places that the actual `st_index` &c rows have real values. This method recomputes the `struct Schema`s to match what's in the system tables.